torch/_inductor/fx_passes/quantization.py

```diff
@@ -660,9 +667,16 @@ def qconv_binary(match: Match, *args, **kwargs):
         accum.realize()
         from .mkldnn_fusion import _can_be_inplace
 
-        assert _can_be_inplace(
-            accum
-        ), "QConv Binary Inplace Fusion requires accum is not an alias or mutation."
+        binary_op_name = binary_unary_attr.binary_op_name
+
+        if binary_op_name == "sum" and (
+            not _can_be_inplace(accum)
+            or accum.data.shape != list(match.nodes[0].meta["val"].size())
+        ):
+            # Change the post op from sum to binary add for two cases:
+            # 1. when outplace add is required.
+            # 2. broadcast add.
+            binary_op_name = "add"
```

Comment:
Same logic here and there. Perhaps we can factor out a util function for it?

Reply:
Done, thanks for your comments!